Framework/Kernel/inc/MantidKernel/SetValueWhenProperty.h

```diff
@@ -0,0 +1,56 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidKernel/IPropertySettings.h"
+#include "MantidKernel/System.h"
+#include <functional>
+#include <iostream>
```

Comment:
Can we remove this header? `<iostream>` should never be needed in a header as it only defines cout/cerr and that's just for debugging. We want to avoid including it as it's very large and adds to compile times.

Reply:
Can the pre-commit checks be configured to catch things like this (like it catches unused imports in python)? It would certainly help me!

Reply:
It's good point. There is a framework called include-what-you-use for C++ but it's considerably more challenging to get going than the python check. Something to consider though...

Reply:
VScode wasn't telling me that these weren't used so I can see that it would be challenging!

Reply:
It's actually a difficult question to answer in C++ so the tooling isn't setup for it